code/lambda.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import print_function
+import sys
+import log_helper
+sys.path.append('lib')
+import numpy
+
+logger = log_helper.getLogger(__name__)
+
+def fv_handler(request, context):
+    # TODO: validate arguments
+
+    logger.info("FV request: {}".format(request))
+    args = [request['rate'], request['nper'], request['pmt'], request.get('pv', 0), request.get('type', 0)]
```

Comment:
I don't see 'type' in the numpy.fv docs. Is this supposed to be 'when'?

Reply:
I'm matching the FV() Excel function interface, rather than numpy. I think we should keep numpy as an implementation detail since we're going to have to implement additional functions as well. I think the target audience is familiar with Excel, so I'm going with that interface.

here's an excerpt of the Excel docs for FV:

```
FV function

FV, one of the financial functions, calculates the future value of an investment based on a constant interest rate. You can use FV with either periodic, constant payments, or a single lump sum payment.

Excel Formula Coach

Use the Excel Formula Coach to find the future value of a series of payments. At the same time, you'll learn how to use the FV function in a formula.

Or, use the Excel Formula Coach to find the future value of a single, lump sum payment.

Syntax

FV(rate,nper,pmt,[pv],[type])

For a more complete description of the arguments in FV and for more information on annuity functions, see PV.

The FV function syntax has the following arguments:

Rate    Required. The interest rate per period.

Nper    Required. The total number of payment periods in an annuity.

Pmt    Required. The payment made each period; it cannot change over the life of the annuity. Typically, pmt contains principal and interest but no other fees or taxes. If pmt is omitted, you must include the pv argument.

Pv    Optional. The present value, or the lump-sum amount that a series of future payments is worth right now. If pv is omitted, it is assumed to be 0 (zero), and you must include the pmt argument.

Type    Optional. The number 0 or 1 and indicates when payments are due. If type is omitted, it is assumed to be 0.
```

Reply:
Got it, makes sense.  Raises an interesting question, though - does that mean all our interfaces will mirror Excel? If so, do we defer to Excel docs for function documentation? Does that then mean we add validation and error handling logic so as not to expose the NumPy exceptions?

Reply:
I think that's the way to go, although let's just start with basic validation (request has the right attributes), rather than going too far to hide NumPy exceptions.